src/Arcus.Messaging.Abstractions/Extensions/IDictionaryExtensions.cs

```diff
@@ -19,7 +19,11 @@ public static class IDictionaryExtensions
         public static (string transactionId, string operationParentId) GetTraceParent(
             this IReadOnlyDictionary<string, object> applicationProperties)
         {
-            Guard.NotNull(applicationProperties, nameof(applicationProperties), "Requires a set of application properties to retrieve the 'traceparent' property");
+            if (applicationProperties is null)
+            {
+                throw new ArgumentNullException(nameof(applicationProperties));
```

Comment:
No `ArgumentNullException.ThrowIfNull` ?

https://learn.microsoft.com/en-us/dotnet/api/system.argumentnullexception.throwifnull?view=net-8.0

Reply:
We talked about this already: not supported in .NET STandard 2.1